client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/thirdpartyclasses/GsonClassReplacement.java

```diff
@@ -7,44 +7,144 @@
 import org.evomaster.client.java.instrumentation.coverage.methodreplacement.UsageFilter;
 import org.evomaster.client.java.instrumentation.object.ClassToSchema;
 
-
+import java.io.IOException;
+import java.io.Reader;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+
 
 public class GsonClassReplacement extends ThirdPartyMethodReplacementClass {
 
+    // TODO: Gson has three more methods, which use their custom classes
+    //  JsonReader and JsonElement.
+    //  Those are not supported yet.
+
     private static final GsonClassReplacement singleton = new GsonClassReplacement();
 
     @Override
     protected String getNameOfThirdPartyTargetClass() {
         return "com.google.gson.Gson";
     }
 
-    // TODO all versions of fromJson
-
     @Replacement(replacingStatic = false,
             type = ReplacementType.TRACKER,
             id = "fromJson_string_class",
             usageFilter = UsageFilter.ANY,
-            category = ReplacementCategory.BASE)
-    public static Object fromJson(Object caller, String json, Class<?> classOfT){
+            category = ReplacementCategory.EXT_0)
+    public static Object fromJson(Object caller, String json, Class<?> classOfT) {
 
-        if(caller == null){
+        if (caller == null) {
             throw new NullPointerException();
         }
 
         ClassToSchema.registerSchemaIfNeeded(classOfT);
-        JsonTaint.handlePossibleJsonTaint(json,classOfT);
+        JsonTaint.handlePossibleJsonTaint(json, classOfT);
 
         Method original = getOriginal(singleton, "fromJson_string_class", caller);
 
         try {
             return original.invoke(caller, json, classOfT);
-        } catch (IllegalAccessException e){
+        } catch (IllegalAccessException e) {
             throw new RuntimeException(e);// ah, the beauty of Java...
-        } catch (InvocationTargetException e){
+        } catch (InvocationTargetException e) {
+            throw (RuntimeException) e.getCause();
+        }
+    }
+
+    @Replacement(
+            replacingStatic = false,
+            type = ReplacementType.TRACKER,
+            id = "fromJson_string_type",
+            usageFilter = UsageFilter.ANY,
+            category = ReplacementCategory.EXT_0
+    )
+    public static Object fromJson(Object caller, String json, Type typeOfT) {
+        if (caller == null) {
+            throw new NullPointerException();
+        }
+
+        Class<?> klass = (Class<?>) typeOfT;
```

Comment:
`Type` is an interface that `Class` extends. There might be other subclasses of `Type` that would wrongly crash here. should maybe rather check if `instanceof`